citystatmodel.cpp

```diff
@@ -110,37 +94,46 @@ bool CityStatModel::setData(const QModelIndex &index, const QVariant &value, int
 {
     if (data(index, role) != value)
     {
+        CityItem &item = m_cityItemsVector[index.row()];
         switch (index.column())
         {
         case 0:
         {
-            m_cityItemsList[index.row()].continentName=value.toString();
+            item.m_continentName = value.toString().toStdString();
             break;
         }
         case 1:
         {
-            m_cityItemsList[index.row()].countryName=value.toString();
+            item.m_countryName = value.toString().toStdString();
             break;
         }
         case 2:
         {
-            m_cityItemsList[index.row()].cityName=value.toString();
+            item.m_cityName = value.toString().toStdString();
             break;
         }
         case 3:
         {
-            m_cityItemsList[index.row()].cityPopulation=value.toUInt();
-            fillPopulationInfo();
+            auto diff = value.toUInt() - item.m_cityPopulation;
```

Comment:
Possible loss of data, you're subtracting a number from unsigned int, what if it is bigger?